numpy/_core/fromnumeric.py

```diff
@@ -2643,6 +2668,35 @@ def all(a, axis=None, out=None, keepdims=np._NoValue, *, where=np._NoValue):
                                   keepdims=keepdims, where=where)
 
 
+def _cumulative_sum_dispatcher(x, /, *, axis=None, dtype=None, out=None,
+                               include_initial=None):
+    return (x, out)
+
+
+@array_function_dispatch(_cumulative_sum_dispatcher)
+def cumulative_sum(x, /, *, axis=None, dtype=None, out=None,
+                   include_initial=False):
+    """
+    Array API compatible cumulative_sum
+    """
+    if out is not None and include_initial:
+        raise ValueError("not supported")
+
+    if x.ndim >= 2 and axis is None:
+        raise ValueError("requires axis arg")
+
+    res = cumsum(x, axis=axis, dtype=dtype, out=out)
+
+    if include_initial:
+        axis = 0 if axis is None else axis
+        initial_shape = list(x.shape)
+        initial_shape[axis] = 1
+        res = np.concat(
+            [np.zeros(initial_shape, dtype=res.dtype), res], axis=axis
```

Comment:
Hmm, seems a bit of a missed opportunity to actually allow `initial` to be passed on too. But easy to adjust if that were to happen.

Reply:
Sure! Let's keep additive and multiplicative identity as `initial` for `cumulative_sum` and `cumulative_prod` for now.

Reply:
If the plan is to update `accumulate`, or add a separate exclusive scan method, I would include this behavior there. I see `cumulative_sum`/`cumsum` as a shorthand for the common `sum.accumulate`, where more advanced cases should be accessed by using the ufunc methods.

And anyways, you can always do `cumulative_sum(x, include_initial=True) + initial`.